diesel/src/macros/mod.rs

```diff
@@ -418,6 +403,10 @@ macro_rules! table_body {
                 }
             }
 
+            impl AppearsInFromClause<table> for table {
```

Comment:
Because they're the same type. This is where we say a table contains itself exactly once. If we can get the changes we need to specialization, I'd have these two blanket impls instead:

```rust
impl<T, U> AppearsInFromClause<T> for U {
    type Count = Never;
}

// The table constraint is to make this impl disjoint from the impl we provide for `Join`
impl<T: Table> AppearsInFromClause<T> for T {
    type Count = Once;
}
```